### PR TITLE
fix: switch input

### PR DESCRIPTION
**Pull request by DorianMaliszewski**
## Type

- Bug
- Enhancement

#### What is expected?

Input should not be under the ball

#### The following changes where made:

1. Ball is behind the input

2. Make some beautiful styles

### CODE REVIEW

File: src/components/Switch/index.tsx [hunk 1]
```diff
@@ -239,16 +266,7 @@ const Switch: VoidFunctionComponent<SwitchProps> = ({
           aria-checked={checked}
           data-variant={variant}
           data-size={size}
-          role="checkbox"
```
Comment (QuiiBz) [inline src/components/Switch/index.tsx:242]: Can we instead use the `switch` role? https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/switch_role

File: src/components/Switch/index.tsx [hunk 2]
```diff
@@ -94,12 +112,18 @@ const StyledSwitch = styled('div', {
   border-radius: 34px;
   position: relative;
   transition: all 250ms;
-  cursor: pointer;
```
Comment (QuiiBz) [inline src/components/Switch/index.tsx:97]: Shouldn't we have a `cursor: pointer` on the whole switch? And `cursor: not-allowed` when disabled

Comment (philibea) [reply src/components/Switch/index.tsx:97]: i'm agree with @QuiiBz, it's weird when you try to click.

https://user-images.githubusercontent.com/14060273/147454947-e4162516-8c63-49e4-a27a-7bf53d26bae4.mov

Comment (vincentaudebert) [reply src/components/Switch/index.tsx:97]: Vu avec Dodo, non necessaire le cursor pointer est deja la.

File: src/components/Switch/index.tsx (hunk 1 above)
Comment (vincentaudebert) [reply src/components/Switch/index.tsx:242]: Vu avec Dodo non necessaire

**Review by vincentaudebert — APPROVED** [review]

**Review by chambo-e — APPROVED** [review]

**Pull request merged**